### ISSUE TITLE
Inconsistency with `incircle`

### ISSUE THREAD

**Issue by lairez** [danshapero/predicates#11]
I have found what looks like inconsistencies with the predicate `incircle`.
You will find in dca6f4c2c75086e6e56ecd343e53d7b8e7221b1d a new unit test that shows the supposed bug.

I construct 4 collinear points whose coordinates are exact integers (in this case, positive integers less that 2^53). `incircle` should output 0.0, but it does not. Then I slightly perturb one of the first three points, so that they define a genuine circle. The sign of the result is still not correct. The correct result can be computed in any computer algebra system.

The original version has similar bugs. Of course I can also be wrong.

**Comment by HansBrende** [comment]
I think the problem lies with your usage of the function `next_after(save, 1)`. I tried reproducing this in a python implementation and your test passes once I replace `nextafter(save, 1)` with `nextafter(save, inf)`. Since `1 < save`, you are returning the next floating point _before_ `save`, not after (i.e., when you say "now we slightly increase the y-coordinate of the second point", you are not increasing it at all, you are decreasing it).

**Comment by lairez** [comment]
You are right, I fixed my usage of nextafter in 380f494095f85bb4cea0dffe2067676e64339bc1.

**However**, the first test (which checks that `incircle` called on four colinear points returns zero) does not use `nextafter` and still failed. So I still feel that there is an inconsistency.
